test/unit/addHooks-test.js

```diff
@@ -83,4 +86,15 @@ describe('addHooks()', () => {
       done(err);
     });
   });
+
+  it('skips hooks loading when there are no hookfiles', (done) => {
+    const transactionRunner = createTransactionRunner();
+    transactionRunner.configuration.hookfiles = [];
+    transactionRunner.configuration.language = 'python';
```

Comment:
Why to test on `python`?
It would spawn a hook handling process, if I'm not mistaken, but does it affect the behavior when no hooks are set?

If the behavior is different depending on whether it's javascript hooks or not, then we would need two tests respectively. But if it's the same, then maybe let's use the default options (javascript)?

Reply:
...like this:

<img width="490" alt="image" src="https://user-images.githubusercontent.com/283441/58179347-4fff6b80-7ca8-11e9-993a-00a73d2defcb.png">

Reply:
But failed attempt to execute not installed hooks handler means the hooks _were_ attempted to run, isn't it?

Reply:
The exception on the screenshot above means the code reached this point:

https://github.com/apiaryio/dredd/blob/0da05fb9c6b0a6a799d5336f9d539eebb9913cf4/lib/addHooks.js#L76-L78

Which, according to the issue, it shouldn't.

Reply:
Correct, because this has been "photographed" at the moment when I didn't have the assert in the test in place and the `if` wasn't in place in the implementation. So it reached the line you pointed out and failed with a try to find Python hooks. That's exactly what happened to @kylef. And that's exactly what shouldn't happen.

When the assert line is in place, it still fails with `configuration` not being undefined. With the `if` in place the test passes.

Reply:
I was first trying to find a failing test, which replicates the erroneous behavior. Then I added the `if`, which fixes it.